.github/workflows/c3-e2e-experimental.yml

```diff
@@ -5,11 +5,6 @@ on:
   pull_request:
     paths:
       - packages/create-cloudflare/**
-  push:
-    branches:
-      - main
-    paths:
-      - packages/create-cloudflare/**
```

Comment:
We don't actually need to run C3 E2E tests on main. They'll be run on Version Packages before release, which has the same effect

Reply:
Why do we want to defer the running of these tests? Won't this get us into a situation where the caretaker is having to debug and fix C3 errors because we haven't tested them correctly?

Reply:
same question as @andyjessop 👍

Reply:
This is the same setup as we have for Wrangler's E2E tests, where we run on Version Packages instead of main. This doesn't actually defer anything, since Version Packages is always in sync with main. Currently we're running C3 E2E tests _twice_ for every commit to main—once on main and once on Version Packages. This PR just removes one of those instances, which should free up some CI capacity.